src/hotspot/share/services/mallocSiteTable.cpp

```diff
@@ -29,7 +29,7 @@
 #include "services/mallocSiteTable.hpp"
 
 // Malloc site hashtable buckets
-MallocSiteHashtableEntry*  MallocSiteTable::_table[MallocSiteTable::table_size];
+MallocSiteHashtableEntry** MallocSiteTable::_table = NULL;
```

Comment:
Reviewer hint: this is a small footprint optimization. Since we initialize NMT only after the VM came up, we can delay allocation of the MallocSizeHashTable and also only do that if NMT is actually on. That way we don't pay for this table if NMT is off, and later could potentially increase its size to decrease the load factor.